pkg/prometheus/promcfg.go

```diff
@@ -1401,6 +1401,10 @@ func (cg *configGenerator) generateRemoteWriteConfig(version semver.Version, p *
 			{Key: "remote_timeout", Value: spec.RemoteTimeout},
 		}
 
+		if len(spec.Headers) > 0 {
```

Comment:
We need to protect this if additionally with the version being >=2.25.0

Reply:
Makes sense. Code is updated.